pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/AuxClassPathController.java

```diff
@@ -0,0 +1,183 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+
+package net.sourceforge.pmd.util.fxdesigner;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.util.List;
+import java.util.ResourceBundle;
+
+import net.sourceforge.pmd.PMDConfiguration;
+import net.sourceforge.pmd.util.ClasspathClassLoader;
+import net.sourceforge.pmd.util.fxdesigner.util.DesignerUtil;
+import net.sourceforge.pmd.util.fxdesigner.util.beans.SettingsOwner;
+
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.fxml.Initializable;
+import javafx.scene.Parent;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.ListView;
+import javafx.stage.FileChooser;
+import javafx.stage.Modality;
+import javafx.stage.Stage;
+
+public class AuxClassPathController implements Initializable, SettingsOwner {
+
+    private final DesignerRoot designerRoot;
+    private ClassLoader classLoader = getClass().getClassLoader();
+
+
+
+    @FXML
+    private Button removeFileButton;
+    @FXML
+    private Button selectFilesButton;
+    @FXML
+    private ListView<File> fileListView;
+    @FXML
+    private Button moveItemUpButton;
+    @FXML
+    private Button moveItemDownButton;
+    @FXML
+    private Button setClassPathButton;
+
+
+    public AuxClassPathController(DesignerRoot designerRoot) {
+        this.designerRoot = designerRoot;
+    }
+
+
+    @Override
+    public void initialize(URL location, ResourceBundle resources) {
+
+        selectFilesButton.setOnAction(e -> onSelectFileClicked());
+        removeFileButton.setOnAction(e -> onRemoveFileClicked());
+        setClassPathButton.setOnAction(e -> {
+            try {
+                setClassPath(classPathGenerator());
+            } catch (IOException e1) {
+                e1.printStackTrace();
+            }
+        });
+        moveItemUpButton.setOnAction(e -> moveUp());
+        moveItemDownButton.setOnAction(e -> moveDown());
+    }
+
+
+    private void onSelectFileClicked() {
+        FileChooser chooser = new FileChooser();
+        chooser.setTitle("Select Files");
+        chooser.getExtensionFilters().addAll(
+            new FileChooser.ExtensionFilter("Java JARs", "*.jar"),
+            new FileChooser.ExtensionFilter("Java WARs", "*.war"),
+            new FileChooser.ExtensionFilter("Java EARs", "*.ear"),
+            new FileChooser.ExtensionFilter("Java class files", "*.class")
+        );
+        List<File> file = chooser.showOpenMultipleDialog(designerRoot.getMainStage());
+        for (File f : file) {
+            fileListView.getItems().add(f);
```

Comment:
you can probably just do `fileListView.getItems().addAll(file)` rather than loop over the list.

Also, since you have a list of files, you probably want to rename the list to `files` rather than the singular `file`